mcstasscript/interface/plotter.py

```diff
@@ -0,0 +1,650 @@
+                min_value = 10.0 ** min_value
+                max_value = 10.0 ** max_value
+            else:
+                
+                min_value = minimum_value 
+                max_value = maximum_value
+
+            # Check the size of the array to be plotted
+            # print(to_plot.shape)
+
+            # Set the axis
+            x_axis_mult = data.plot_options.x_limit_multiplier
+            y_axis_mult = data.plot_options.y_limit_multiplier
+
+            X = np.linspace(data.metadata.limits[0]*x_axis_mult,
+                            data.metadata.limits[1]*x_axis_mult,
+                            data.metadata.dimension[0]+1)
+            Y = np.linspace(data.metadata.limits[2]*y_axis_mult,
+                            data.metadata.limits[3]*y_axis_mult,
+                            data.metadata.dimension[1]+1)
+
+            # Create a meshgrid for both x and y
+            x, y = np.meshgrid(X, Y)
+
+            # Generate information on necessary colorrange
+            levels = MaxNLocator(nbins=150).tick_values(min_value,
+                                                        max_value)
+
+            # Select colormap
+            cmap = plt.get_cmap(data.plot_options.colormap)
+
+            # Select the colorscale normalization
+            if data.plot_options.log:
+                norm = matplotlib.colors.LogNorm(vmin=min_value,
+                                                 vmax=max_value)
+            else:
+                norm = BoundaryNorm(levels, ncolors=cmap.N, clip=True)
+
+            # Create plot
+            im = ax.pcolormesh(x, y, Intensity,
+                               cmap=cmap, norm=norm)
+
+            def fmt(x, pos):
+                a, b = '{:.2e}'.format(x).split('e')
+                b = int(b)
+                if abs(float(a) - 1) < 0.01 :
+                    return r'$10^{{{}}}$'.format(b)
+                else:
+                    return r'${} \times 10^{{{}}}$'.format(a, b)
+
+            # Add the colorbar
+            if data.plot_options.colormap:
+                fig.colorbar(im, ax=ax,
+                             format=matplotlib.ticker.FuncFormatter(fmt))
+
+            # Add a title
+            ax.set_title(data.metadata.title)
+
+            # Add axis labels
+            ax.set_xlabel(data.metadata.xlabel)
+            ax.set_ylabel(data.metadata.ylabel)
+
+            if data.plot_options.custom_ylim_top:
+                ax.set_ylim(top=data.plot_options.top_lim)
+
+            if data.plot_options.custom_ylim_bottom:
+                ax.set_ylim(bottom=data.plot_options.bottom_lim)
+
+            if data.plot_options.custom_xlim_left:
+                ax.set_xlim(left=data.plot_options.left_lim)
+
+            if data.plot_options.custom_xlim_right:
+                ax.set_xlim(right=data.plot_options.right_lim)
+
+        def init_1D():
+            # initialize function for animation
+            er.set_data([], [], []) # wont work
```

Comment:
I would be very careful using functions inside function to access the scope of the `__init__`. I would either
- pass things as arguments (e.g. `er` needs to be an input argument and this function needs to be outside of the class)
- or use the fact that we are inside a class, and make `er` a member of the class (`self.er = ...`) and then make these functions class members, i.e.
```Py
def init_1D(self):
    self.er = set_data([], [], [])
```

Reply:
The FuncAnimation need a function for updating the data with only an index as argument (unless I am misunderstanding), and in such I need to define a short function inside this function. I find it acceptable since the function is so short.